experimental/packages/opentelemetry-api-metrics/src/types/Metric.ts

```diff
@@ -61,25 +61,25 @@ export enum ValueType {
  *   <li> count the number of 5xx errors. </li>
  * <ol>
  */
-export interface Counter {
+export interface Counter<T extends string> {
```

Comment:
It may be worth to allow specifying the value types too. Also, I'd suggest making the type parameter optional so that people don't have to set the type parameters if they don't need to.

```suggestion
export interface Counter<AttributesType extends MetricAttributes = MetricAttributes> {
```

With this type definition, people can define their attributes type like: 
```ts
interface MyAttributes {
  foo: number;
  bar: string;
}

meter.createCounter<MyAttributes>('my-counter');
// or, without the type paramter.
meter.createCounter('my-counter');
```

Reply:
Another point worth mentioning on @legendecas suggestion is giving the type a human readable name is usually better than `T`